elastic/kibana
# [Fleet] Add assets tab

## Summary

Add the "Assets" tab to the integration details view.

## How to test

1. Start Kibana + ES on basic
2. Go to fleet and follow the in-app setup instructions for getting an agent set up
3. Go to integrations
4. Choose an integration to add from the "browse" view
5. Note: there should be no "Assets" tab visible yet. Select "Add <package name>", 
6. Complete the form that just loaded (if necessary) and click "Save integration"
7. You should be redirected to the integrations UI (where you were before the form) with a toast containing a link
8. Click the link showing the "Add agent" flyout
9. The policy should be pre-selected (i.e., no policy selection option in the flyout, just an option to choose the enrollment token)
10. At the bottom of the flyout there should be a button to "View assets"
11. Clicking "View assets" should take you to the new "Assets" tab
12. (Optional) Also test clicking the "+" button when viewing the policies for an integration. This should open the same flyout as when clicking the success toast described above.

## To reviewers

* There is risk with this implementation that we could link to UI that does not exist/is disabled. Specifically dashboards, visualisations and discover. It seems unlikely, but the current implementation has no way of accounting for this which creates some risk. The problem already existed (see comments and restructuring of `x-pack/plugins/fleet/public/hooks/use_kibana_link.ts`).
* Any ideas for adding tests would be appreciated! I think this is already a lot of new code to digest in a review. Ultimately I think we need functional tests to give us good coverage. I'd like to take this on in a follow up PR though if possible which means thorough manual testing is imperative!

## Screenshots

### Package not yet added (should be the same as before since we have not installed the package)

![Screenshot 2021-06-17 at 16 35 02](https://user-images.githubusercontent.com/8155004/122418955-e0453300-cf8a-11eb-80e3-fdefeeade80d.png)

### Success toast after adding a package with link

![Screenshot 2021-06-17 at 16 34 37](https://user-images.githubusercontent.com/8155004/122419036-ef2be580-cf8a-11eb-8ae9-ae04a77f2652.png)

### New flyout with "Add assets link"

![Screenshot 2021-06-17 at 16 34 50](https://user-images.githubusercontent.com/8155004/122419150-04087900-cf8b-11eb-800e-1a0ec2646e27.png)

### New "Assets" tab with sections expanded and links implemented

<img width="1315" alt="Screenshot 2021-06-18 at 12 05 58" src="https://user-images.githubusercontent.com/8155004/122544953-95302c00-d02d-11eb-881c-59fef4b8adb0.png">

### Checklist




### Risk Matrix

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Some refactoring and introducing of links to external UI | Med | Med | This needs to be mitigated by using the new URL service |
